### PR TITLE
#2843, #2806

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2843 Failure to set `AfterHostStarted` working directory stops RHost
- add ACL check on the directory

#2806 Still occasionally get package load failures
- store provider rather than create workflow when editor instantiates keyboard processor

#2853 Error adding database connection
- reference signed and versioned binaries from lib

Update System.Collections.Immutable to 1.3.0

### CODE REVIEW

File: src/Common/Core/Impl/Extensions/IOExtensions.cs [hunk 1]
```diff
@@ -95,5 +97,33 @@ public static class IOExtensions {
             }
             return Path.DirectorySeparatorChar.ToString();
         }
+
+        public static bool HasWritePermission(this string path) {
```
Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: OK. I wanted stricter check since writing generally assumes reading (in common sense case anyway)

Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: OK. I wanted stricter check since writing generally assumes reading (in common sense case anyway)

Comment (int19h) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: I don't think that is the case for Windows ACLs, at least - I just tried, and you can deny read and allow write to the same user at the same time, and it actually enforces it in that manner.

Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: Yes, it is possible in general case since ACLs may apply to, say, pipes.

Comment (MikhailArkhipov) [inline src/Common/Core/Impl/Extensions/IOExtensions.cs]: I'll leave writing check method for now in case we need it

**Pull request merged**